### PR TITLE
added test case: drop_publication_stops_the_replication_on_subscriber

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [ ] Touched code is covered by tests
 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/integrationtests/MetadataTrackerITest.java [hunk 1]
```diff
@@ -297,6 +299,35 @@ public void test_dropped_table_is_replicated() throws Exception {
...
+                    "SELECT s.subname, sr.srrelid::text, sr.srsubstate, sr.srsubstate_reason" +
+                        " FROM pg_subscription s" +
+                        " LEFT JOIN pg_subscription_rel sr ON s.oid = sr.srsubid" +
+                        " ORDER BY s.subname");
+                assertThat(printedTable(res.rows()), CoreMatchers.is("sub1| doc.t1| e| Tracking of metadata failed for subscription 'sub1' with unrecoverable error, stop tracking.\n" +
+                    "Reason: Publication 'pub1' unknown\n"));
+            }, 20, TimeUnit.SECONDS
+        );
+
+        response = executeOnSubscriber("INSERT INTO t1 (id) VALUES(3)");
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/integrationtests/MetadataTrackerITest.java:327]: this assertion currently fails, i.e tracking is stopped but table is still "locked".

Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/MetadataTrackerITest.java:327]: I think this is actually okay. Users will have to take a look at what happened by checking the subscription tables/logs, and if they're okay with the publication being gone they can drop the subscription and after that it should work(?)

Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/MetadataTrackerITest.java:327]: thx for the feedback! I will check with drop sub - if it works I will update documents - we already say that replication stops but will add a note that table is read-only and one have to use DROP SUB

Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/MetadataTrackerITest.java:336]: > probably need this test case?

yep I think it is good to keep it

**Pull request merged**